syndeo-template/src/test/java/com/google/cloud/syndeo/it/KafkaToBigQueryIT.java

```diff
@@ -0,0 +1,291 @@
+    if (!DataflowUtils.waitUntilState(
+        dataGeneratorPipeline, PipelineResult.State.RUNNING, SIX_MINUTES_MILLIS)) {
+      dataGeneratorPipeline.cancel();
+      throw new RuntimeException(
+          "Data generator pipeline did not start by deadline. Cancelling test");
+    }
+
+    // Start syndeo pipeline.
+    FlexTemplateClient templateClient =
+        FlexTemplateClient.builder().setCredentials(CREDENTIALS).build();
+    {
+      DataflowClient.JobInfo syndeoPipeline = kickstartSyndeoPipeline();
+      DataflowOperator operator = new DataflowOperator(templateClient);
+
+      templateClient.waitUntilActive(PROJECT, REGION, syndeoPipeline.jobId());
+
+      // Wait for two minutes while the pipeline executes, and then drain it.
+      operator.waitUntilDoneAndFinish(
+          DataflowOperator.Config.builder()
+              .setProject(PROJECT)
+              .setRegion(REGION)
+              .setJobId(syndeoPipeline.jobId())
+              .setTimeoutAfter(java.time.Duration.ofMinutes(3))
+              .build());
+    }
+
+    // Start a new syndeo pipeline. We do this to test for appropriate behavior on restarts
+    DataflowClient.JobInfo syndeoPipeline = kickstartSyndeoPipeline();
+    DataflowOperator operator = new DataflowOperator(templateClient);
+
+    templateClient.waitUntilActive(PROJECT, REGION, syndeoPipeline.jobId());
+    // Sleep while the pipeline runs to move all the data.
+    Thread.sleep(3 * ONE_MINUTE_MILLIS);
+
+    // Wait for a while to check BigQuery results
+    {
+      Instant start = Instant.now();
+      while (true) {
+        Long rowsInTable =
+            bigQueryResourceManager.getRowCount(
+                PROJECT,
+                BIGQUERY_DATASET,
+                BIGQUERY_TABLE.substring(BIGQUERY_TABLE.lastIndexOf(".") + 1));
+        if (rowsInTable == ELEMENTS_PER_SECOND * PIPELINE_DURATION_SECONDS) {
+          // Test passes yay!
+          break;
+        } else if (rowsInTable > ELEMENTS_PER_SECOND * PIPELINE_DURATION_SECONDS) {
+          throw new AssertionError(
+              String.format(
+                  "Expected at most %s elements to be inserted to BigQuery, but found %s.",
+                  ELEMENTS_PER_SECOND * PIPELINE_DURATION_SECONDS, rowsInTable));
+        } else if (java.time.Duration.between(start, Instant.now())
+                .compareTo(java.time.Duration.ofMinutes(20))
+            > 0) {
+          // If we spent over 20 minutees waiting for the job, then we must exit
+          throw new AssertionError(
+              String.format(
+                  "Expected %s elements to be inserted to BigQuery, but found %s after over 20 minutes.",
+                  ELEMENTS_PER_SECOND * PIPELINE_DURATION_SECONDS, rowsInTable));
+        } else {
+          // Iterate once more
+          Thread.sleep(ONE_MINUTE_MILLIS);
+          continue;
+        }
+      }
+    }
+
+    templateClient.drainJob(PROJECT, REGION, syndeoPipeline.jobId());
```

Comment:
this is not needed, the job is drained when `drainJobAndFinish` is called below

Reply:
great catch thank pranav : )